app/controllers/books_controller.rb

```diff
@@ -2,6 +2,14 @@ class BooksController < ApplicationController
   before_filter :authenticate_user!
 
   def index
-    @books = Book.all
+    sort = params[:sort] || "published_date desc"
+    @books = Book.filter(params[:search])
+      .order(sort)
+      .page(params[:page])
```

Comment:
I feel like there's an opinionated way on how to do searches in the controller but I don't know what that way is. I think your way is DRY and gucci af.